### ISSUE TITLE
SqlException when port specified at ConnectionString

### ISSUE THREAD

**Issue by SLavrynenko** [StackExchange/Dapper#967]
Seems that Dapper doesnt support connection string with port specified. So I've following connection string:

        <add name="SFConnection" connectionString="Server=mysqlserver,5001;Initial Catalog=XXX;Integrated Security=False;User Id=sa;Password=YYYY" providerName="System.Data.SqlClient" />

and I've got sql exception while trying to query some client from the database:

`return connection.Query<ClientInfo>(SelectClientsSqlString, new { Ids = clientIds }).ToList();`

Exception:
> 		Message	"Incorrect syntax near ','."	string
> 		Number	102	int
> 		Procedure	""	string
> 		Server	"mysqlserver,5001"	string
> 		Source	".Net SqlClient Data Provider"	string
> 		StackTrace	"   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)\r\n   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)\r\n   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)\r\n   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)\r\n   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()\r\n   at System.Data.SqlClient.SqlDataReader.get_MetaData()\r\n   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption)\r\n   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)\r\n   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)\r\n   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)\r\n   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)\r\n   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)\r\n   at System.Data.Common.DbCommand.System.Data.IDbCommand.ExecuteReader(CommandBehavior behavior)\r\n   at Dapper.SqlMapper.ExecuteReaderWithFlagsFallback(IDbCommand cmd, Boolean wasClosed, CommandBehavior behavior)\r\n   at Dapper.SqlMapper.<QueryImpl>d__136`1.MoveNext()\r\n   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)\r\n   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)\r\n   at Dapper.SqlMapper.Query[T](IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Boolean buffered, Nullable`1 commandTimeout, Nullable`1 commandType)\r\n   at XXX.GetClients(IEnumerable`1 clientIds) in > 

I've the last version of Dapper:

>   package id="Dapper" version="1.50.4" targetFramework="net451" 

Is there any workaround to make this work?

**Comment by mgravell** [comment]
Dapper doesn't touch the port or any other connection concept. It takes the connection you give it. It leterally has no reason to care.

So: what is `SelectClientsSqlString` here? It sounds like it isn't valid SQL. I don't think it has anything to do with ports.

**Comment by SLavrynenko** [comment]
Here is the command:

`        private const string SelectClientsSqlString = "SELECT ClientId, ClientName from XXX where ClientId In (@Ids)";
`

**Comment by mgravell** [comment]
K, I don't have access to your setup, so my advice would be: simplify it to see what makes it break. Try removing the "in" clause (perhaps just use an equality filter for now), or try selecting a single column - just to.narrow down when it is failing. When we know what exactly makes it fail: it should be easier to suggest workarounds.

**Comment by SLavrynenko** [comment]
@mgravell , so I've found what was the root cause. The correct sql command here is:

```
       private const string SelectClientsSqlString = "SELECT ClientId, ClientName from ClientsCentral where ClientId In @Ids";

```

I dont know why it doesnt like brackets around @Ids, but now it works. Sorry for false alarm, but I was confused by error message containing that there is something wrong after the comma:

>  Message	"Incorrect syntax near ','."	string
>  Server	"mysqlserver,5001"	string

So I decided that it doesnt like port number in connection string. Sorry again
